### PR TITLE
Switch lists of lists validation to be a warning

### LINKED ISSUE #3905: Lists of lists not supported
**Describe the bug**
I have a cypher query that returns a List of lists. When I try to upgrade from version 3.x to 4.0, I get the following error:

```
Failed to get GraphQL schema: [
  GraphQLError: Invalid field type: Lists of lists are not supported.
      at createGraphQLError (/Users/jansen/javascript/graphbiq-api/node_modules/@neo4j/graphql/dist/schema/validation/custom-rules/utils/document-validation-error.js:36:12)
      at Object.FieldDefinition (/Users/jansen/javascript/graphbiq-api/node_modules/@neo4j/graphql/dist/schema/validation/custom-rules/valid-types/valid-field-types.js:49:88)
      at Object.enter (/Users/jansen/javascript/graphbiq-api/node_modules/graphql/language/visitor.js:301:32)
      at visit (/Users/jansen/javascript/graphbiq-api/node_modules/graphql/language/visitor.js:197:21)
      at validateSDL (/Users/jansen/javascript/graphbiq-api/node_modules/@neo4j/graphql/dist/schema/validation/validate-sdl.js:32:25)
      at runValidationRulesOnFilteredDocument (/Users/jansen/javascript/graphbiq-api/node_modules/@neo4j/graphql/dist/schema/validation/validate-document.js:177:51)
      at validateDocument (/Users/jansen/javascript/graphbiq-api/node_modules/@neo4j/graphql/dist/schema/validation/validate-document.js:219:5)
      at /Users/jansen/javascript/graphbiq-api/node_modules/@neo4j/graphql/dist/classes/Neo4jGraphQL.js:267:51
      at new Promise (<anonymous>)
      at Neo4jGraphQL.generateExecutableSchema (/Users/jansen/javascript/graphbiq-api/node_modules/@neo4j/graphql/dist/classes/Neo4jGraphQL.js:263:16) {
    path: [ 'pathList', 'paths' ],
    locations: [ [Object] ],
    extensions: [Object: null prototype] {}
  }
]
```

**Type definitions**
```type Entity  @mutation(operations: [])  {
    paths: pathList!
               @cypher(
                       statement: """
                       WITH [[{entity_id:1, other_entity_id: 2}],[{entity_id: 2, other_entity_id:3},{entity_id:3, other_entity_id: 4}]] as paths
                       RETURN {paths: paths} as result
                       """, columnName: "result"
                   )
}
type pathList @query(read: false, aggregate: false) @mutation(operations: []) @subscription(events: []) {
    paths: [[pathLink]]
}
type pathLink @query(read: false, aggregate: false) @mutation(operations: []) @subscription(events: []){
  entity_id: Int
  other_entity_id: Int
}
```

**To Reproduce**
1. Put the type definitions in a schema, create an index file as in  the 'getting started' example and try 'node index.js'

**Expected behavior**
I am expecting the schema to be able to handle query that returns a list of lists. Version 3.x was able to, no problem, so this is a breaking change, but is has not been documented in the migration guide.

**System (please complete the following information):**

* OS: MacOS 13.4.1
* Version: @neo4j/graphql@4.1.1
* Node.js version: v20.5.1

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

Switches lists of lists validation to be a warning, to allow them to be used as the result of custom Cypher/resolvers.

Also fixes a small bug where lists with non-null modifiers were being ignored.

## Complexity

Complexity: Low

# Issue

Closes #3905

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3937.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 34 | 46 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 73 | 93 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 51 | 56 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3049 | 3428 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2980 | 3026 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 57 | 56 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 42 | 55 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 13070331 | 4823 | 5306 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 150 | 265 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 101 | 122 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 201 | 220 | 3003 |
|  connections.Connection | 14082 | 14082 | 63 | 69 | 2174 |
|  connections.NestedConnection | 45470 | 45470 | 118 | 132 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 70 | 95 | 1040 |
|  connections.ConnectionWithSortAndCypher | 16325 | 16325 | 105 | 179 | 2174 |
|  create.SimpleMutation | 7 | 7 | 61 | 100 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 38 | 35 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 307 | 260 | 1040 |
|  delete.NestedDeleteInUpdate | 19198 | 19198 | 262 | 220 | 1179 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 62 | 62 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 117 | 113 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 41 | 56 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 77 | 92 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 22 | 24 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 48 | 61 | 2174 |
|  query.QueryWhere | 9704 | 9706 | 38 | 44 | 2163 |
|  query.SimpleQueryWithNestedWhere | 9882 | 9884 | 55 | 63 | 2163 |
|  query.Nested | 10096041 | 10096041 | 10658 | 10879 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10747 | 11071 | 2004000 |
|  query.OrFilterOnRelationships | 40757 | 42744 | 201 | 263 | 1224 |
|  query.OrFilterOnRelationshipsAndNested | 34671 | 37323 | 227 | 329 | 1224 |
|  query.QueryWithNestedIn | 13345 | 13739 | 62 | 82 | 1198 |
|  query.NestedConnectionWhere | 9834 | 9834 | 60 | 119 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10054 | 9954 | 81 | 248 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19350 | 19128 | 213 | 234 | 1664 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 257 | 248 | 1134 |
|  query.Fulltext | 80 | 80 | 43 | 39 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 46 | 56 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 77 | 104 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14501 | 14520 | 107 | 126 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 53 | 80 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 102 | 114 | 4516 |
|  sorting.SortWithTopLevelCypher | 3396 | 3396 | 64 | 73 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 69 | 87 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 63 | 68 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 288 | 316 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 328 | 424 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 202 | 114 | 2002 |

</details>
        
Old Schema Generation: 44.994s
Schema Generation: 45.371s
Old Subgraph Schema Generation: 55.798s
Subgraph Schema Generation: 54.829s

File: packages/graphql/src/schema/validation/custom-rules/warnings/list-of-lists.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    }
+                    // const isListOfLists = listNode.type.kind === Kind.LIST_TYPE;
+                    // // TODO: figure this out - seems to have no impact having this commented-out
+                    // // && listNode.type.type.kind === Kind.LIST_TYPE;
+                    // if (isListOfLists) {
+                    //     console.warn(
+                    //         "Encountered list field definition(s) with list elements. This is not supported by Neo4j, however, you can ignore this warning if the field is only used in the result of custom resolver/Cypher."
+                    //     );
+                    //     warningAlreadyIssued = true;
+                    // }
```
Comment (a-alle) [inline packages/graphql/src/schema/validation/custom-rules/warnings/list-of-lists.ts:46]: is this comment useful?

**Review by a-alle — DISMISSED** [review]

File: packages/graphql/src/schema/validation/custom-rules/warnings/list-of-lists.ts (hunk 1 above)
Comment (darrellwarde) [reply packages/graphql/src/schema/validation/custom-rules/warnings/list-of-lists.ts]: It was already there in the original implementation! 😆 But I've removed it now.

**Pull request merged**